src/contract.ts

```diff
@@ -0,0 +1,381 @@
+      } else {
+        return this.call(fragment.name, args);
+      }
+    };
+  }
+
+  /**
+   *
+   * @param name
+   * @param params
+   * @param accounts
+   * @param writableAccounts
+   * @param seeds
+   * @param signers
+   * @param caller
+   * @returns
+   */
+  private async call(
+    name: string,
+    args: any[],
+    options?: ContractTransactionOptions
+  ): Promise<ContractCallResult> {
+    const {
+      accounts = [],
+      writableAccounts = [],
+      seeds = [],
+      signers = [],
+      caller = this.program.payerAccount.publicKey,
+      value = 0,
+      simulate = false,
+    } = options ?? {};
+
+    const fragment = this.abi.getFunction(name);
+    const input = this.abi.encodeFunctionData(name, args);
+
+    const data = Buffer.concat([
+      this.contractStorageAccount.publicKey.toBuffer(),
+      caller.toBuffer(),
+      Buffer.from(numToPaddedHex(value), 'hex'),
+      Buffer.from('00000000', 'hex'),
+      encodeSeeds(seeds),
+      Buffer.from(input.replace('0x', ''), 'hex'),
+    ]);
+
+    const keys = [
+      ...seeds.map((seed) => ({
+        pubkey: seed.address,
+        isSigner: false,
+        isWritable: true,
+      })),
+      {
+        pubkey: this.contractStorageAccount.publicKey,
+        isSigner: false,
+        isWritable: true,
+      },
+      {
+        pubkey: SYSVAR_CLOCK_PUBKEY,
+        isSigner: false,
+        isWritable: false,
+      },
+      {
+        pubkey: PublicKey.default,
+        isSigner: false,
+        isWritable: false,
+      },
+      ...accounts.map((pubkey) => ({
+        pubkey,
+        isSigner: false,
+        isWritable: false,
+      })),
+      ...writableAccounts.map((pubkey) => ({
+        pubkey,
+        isSigner: false,
+        isWritable: true,
+      })),
+    ];
+
+    const instruction = new TransactionInstruction({
+      keys,
+      programId: this.program.programAccount.publicKey,
+      data,
+    });
+
+    signers.unshift(this.program.payerAccount);
+
+    const { encoded, logs, computeUnitsUsed } = await this.program.makeTx(
+      simulate,
+      instruction,
+      signers
+    );
+
+    let result: ethers.utils.Result | null = null;
+
+    if (fragment.outputs?.length) {
+      if (!encoded) {
+        throw new Error('return data not set');
+      }
+
+      result = this.abi.decodeFunctionResult(fragment, encoded);
+      if (fragment.outputs.length === 1) {
+        result = result[0];
+      }
+    }
+
+    const events = this.parseLogsEvents(logs);
+
+    return { result, logs, computeUnitsUsed, events };
+  }
+
+  /**
+   *
+   * @param test
+   * @param upto
+   * @returns
+   */
+  async contractStorage(test: Program, upto: number): Promise<Buffer> {
+    const accountInfo = await test.connection.getAccountInfo(
+      this.contractStorageAccount.publicKey
+    );
+
+    return accountInfo!.data;
+  }
+
+  /**
+   *
+   * @returns
+   */
+  getStorageKeyPair(): Keypair {
+    return this.contractStorageAccount;
+  }
```

Comment:
A getter with the same name as the property would produce a more obvious API. Whether you call things accounts or keypairs, it should be consistent.